src/component.ts

```diff
@@ -70,3 +58,25 @@ export function componentFactory (
     : Vue
   return Super.extend(options)
 }
+
+function applyDecorators (
+  options: Vue.ComponentOptions<Vue>,
+  Component: VueClass,
+  meta: Meta
+): void {
+  if (!meta.shouldGetInitalProperty) {
+    forEachValues(meta.decoratorMap, fn => {
+      fn(options)
+    })
+    return
+  }
+
+  // Prevent to run _init function of Vue
+  // because we just want to get initial properties.
+  Component.prototype._init = noop
+  const data = new Component()
```

Comment:
Caveat: this means constructor will always be called on component definition. This will be a source of gotchas for side-effectful initializers.